### ISSUE TITLE
Where do excerpts for old posts live?

### ISSUE THREAD

**Issue by dreeves** [beeminder/blog#377]
[placeholder]

**Comment by dreeves** [comment]
In posts.json, the excerpt for many posts are blank. But they apparently have excerpts:

<img width="541" alt="image" src="https://github.com/beeminder/blog/assets/9928/db4ebf59-7ac7-4923-b98e-f3cfd871f5ae">

Are those auto-generated from the post itself or are they stored somewhere?

(I can find this out myself by making a small change to the beginning of a post, republishing, and seeing if the excerpt changes.)

**Comment by narthur** [comment]
@dreeves Yes. Currently the blog auto-generates excerpts for legacy posts which don't have one. The code that does it:

[src/lib/getExcerpt.ts](https://github.com/beeminder/blog/blob/master/src/lib/getExcerpt.ts)

**Comment by dreeves** [comment]
Got it! Possibly this should be part of #380 now but I believe that it's bad to treat "" as the special indicator for autogenerating an excerpt from the post. We should have an explicit magic string for that. If you give "" as the explicit excerpt, that should mean you want the empty string as the excerpt. And to be extreme anti-magic, it should be an error if you omit the excerpt field altogether.

**Comment by narthur** [comment]
@dreeves For sure. I think that may have been an artifact of the code I wrote to migrate legacy data to posts.json. If we wanted, I could do something similar, take all those auto excerpts and put them in posts.json, and then delete the automatic excerpt functionality.

**Comment by dreeves** [comment]
Top-level description updated!

**Comment by narthur** [comment]
@dreeves Looks good, though I think this issue's description is out-of-date now.

**Comment by narthur** [comment]
@dreeves I believe we've gone through posts.json now and explicitly set the auto excerpt flag for them with merge of #415
